pinot-controller/src/main/java/com/linkedin/pinot/controller/ControllerStarter.java

```diff
@@ -152,6 +153,9 @@ public void start() {
       LOGGER.info("Starting segment status manager");
       _segmentStatusChecker.start();
 
+      LOGGER.info("Creating rebalance segments factory");
+      RebalanceSegmentsFactory.createInstance(_helixResourceManager.getHelixZkManager());
```

Comment:
This will require us to restart the controller in case we change the segment assignment strategy since `RebalanceSegmentsFactory` is a singleton class. Do you prefer to leave this way or do you think it's better to create rebalance strategy every time we need to use?